### PR TITLE
Support try expr

### PR DESCRIPTION
**Pull request by amutake**
I implemented:

- try-catch expression
- try-of-catch expression
- try-after expression
- try-of-after expression
- try-catch-after expression
- try-of-catch-after expression
- catch clause
- catch clause with exception class
- catch clause with exception class and stacktrace
- catch clause with guard sequence
- catch clause with exception class and guard sequence
- catch clause with exception class, stacktrace, and guard sequence

This PR has >1000 line changes, but almost all changes are test code.

### CODE REVIEW

File: lib/abstract_format.ml [hunk 1]
```diff
@@ -70,6 +70,7 @@ and expr_t =
   | ExprMatch of {line: line_t; pattern: pattern_t; body: expr_t}
   | ExprBinOp of {line: line_t; op: string; lhs: expr_t; rhs: expr_t}
   | ExprTuple of {line: line_t; elements: expr_t list}
+  | ExprTry of {line: line_t; exprs: expr_t list; case_clauses: clause_t list; catch_clauses: clause_t list; after: expr_t list}
```
Comment (amutake) [inline lib/abstract_format.ml]: I unified all `try`-expressions as a variant.

File: lib/abstract_format.ml [hunk 2]
```diff
@@ -87,6 +88,9 @@ and integer_or_var_t =
 
 and clause_t =
   | ClsCase of {line: line_t; pattern: pattern_t; guard_sequence: guard_sequence_t option; body: expr_t}
+  | ClsCatch of {line: line_t; line_cls: line_t; line_stacktrace: line_t;
```
Comment (amutake) [inline lib/abstract_format.ml]: I unified all `catch`-clause as a variant.

**Comment by amutake** [comment]
Oh... I must support OTP-19 and OTP-20.

File: test/Makefile [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
 
 .PHONY: compile
-compile: $(beams)
+compile: $(beams) otp_version
 
 %.beam: %.erl
 	erlc +debug_info -o $@ $<
 
+otp_version:
+	erl +V 2>&1 | awk '{ print $$NF }' | awk -F. '{ print $$1 }' | xargs expr 11 + | tr -d '\n' > otp_version
```
Comment (amutake) [inline test/Makefile]: I made it possible to save Erlang/OTP version into `otp_version` file on build time of beam files.
The `otp_version` file is used in version-dependent tests (e.g., `test_try_stacktrace`).

**Comment by amutake** [comment]
@yutopp Could you review this PR? I added support of OTP-19, 20, and 21.

**Comment by yutopp** [comment]
I also updated https://github.com/yutopp/obeam/issues/54.

**Comment by amutake** [comment]
Thank you for your review!

**Pull request merged**